ddev/src/ddev/cli/meta/scripts/upgrade_python.py

```diff
@@ -3,230 +3,391 @@
-        if ruff_config.get('target-version') == old_version:
-            ruff_config['target-version'] = new_version
-            tracker.success()
-            changed = True
+    Returns:
+        True if successful, False otherwise
+    """
+    try:
+        file_path.write_text(content)
+        tracker.success()
+        return True
+    except Exception as e:
+        tracker.error((file_label,), message=f'Failed to write: {e}')
+        return False
 
-    if changed:
-        config_file.write_text(tomlkit.dumps(config))
 
+def extract_sha256_from_sbom(packages: list, platform_name: str) -> str:
+    """
+    Extract SHA256 hash from SBOM packages.
 
-def update_setup_file(target, new_version: str, old_version: str, tracker: ValidationTracker):
-    setup_file = target.path / 'setup.py'
+    Args:
+        packages: List of packages from SBOM
+        platform_name: Platform name for error messages (e.g., "Linux", "Windows")
 
-    if setup_file.is_file():
-        content = setup_file.read_text()
+    Returns:
+        SHA256 hash string
 
-        if f"Programming Language :: Python :: {old_version}" in content:
-            content = content.replace(
-                f"Programming Language :: Python :: {old_version}", f"Programming Language :: Python :: {new_version}"
-            )
+    Raises:
+        ValueError: If package, checksum, or hash format is invalid
+    """
+    cpython_package = next((pkg for pkg in packages if pkg.get('name') == "CPython"), None)
+    if cpython_package is None:
+        raise ValueError(f"Could not find CPython package in {platform_name} SBOM")
+
+    checksums = cpython_package.get('checksums', [])
+    checksum = next((cs for cs in checksums if cs.get('algorithm') == "SHA256"), None)
+    if checksum is None:
+        raise ValueError(f"Could not find SHA256 checksum in {platform_name} SBOM")
+
+    hash_value = checksum.get('checksumValue', '')
+    if not validate_sha256(hash_value):
+        raise ValueError(f"Invalid {platform_name} SHA256 hash format from SBOM: {hash_value}")
+
+    return hash_value
+
+
+def upgrade_dockerfiles_python_version(
+    app: Application, new_version: str, hashes: dict[str, str], tracker: ValidationTracker
+):
+    dockerfiles = [
+        app.repo.path / '.builders' / 'images' / 'linux-aarch64' / 'Dockerfile',
+        app.repo.path / '.builders' / 'images' / 'linux-x86_64' / 'Dockerfile',
+        app.repo.path / '.builders' / 'images' / 'windows-x86_64' / 'Dockerfile',
+    ]
+
+    try:
+        linux_sha = hashes['linux_source_sha256']
+        windows_sha = hashes['windows_amd64_sha256']
+    except KeyError as error:
+        tracker.error(('Dockerfiles',), message=f'Missing SHA256 hash entry: {error}')
+        return
 
-            setup_file.write_text(content)
-            tracker.success()
+    for dockerfile in dockerfiles:
+        content = read_file_safely(dockerfile, dockerfile.name, tracker)
```

Comment:
nit. Can the variable content here be a bit more descriptive? Like dockerfile_contents?